Sources/GRPCInterceptors/Tracing/SpanAttributes+RPCAttributes.swift

```diff
@@ -0,0 +1,209 @@
+    serverHostname: String,
+    networkTransportMethod: String
+  ) {
+    self.attributes.rpc.system = "grpc"
+    self.attributes.rpc.serverAddress = serverHostname
+    self.attributes.rpc.networkTransport = networkTransportMethod
+    self.attributes.rpc.service = context.descriptor.service.fullyQualifiedService
+    self.attributes.rpc.method = context.descriptor.method
+
+    // Set server address information
+    switch PeerAddress(context.localPeer) {
+    case .ipv4(let address, let port):
+      self.attributes.rpc.networkType = "ipv4"
+      self.attributes.rpc.networkPeerAddress = address
+      self.attributes.rpc.networkPeerPort = port
+      self.attributes.rpc.serverPort = port
+
+    case .ipv6(let address, let port):
+      self.attributes.rpc.networkType = "ipv6"
+      self.attributes.rpc.networkPeerAddress = address
+      self.attributes.rpc.networkPeerPort = port
+      self.attributes.rpc.serverPort = port
+
+    case .unixDomainSocket(let path):
+      self.attributes.rpc.networkType = "unix"
+      self.attributes.rpc.networkPeerAddress = path
+
+    case .other(let address):
+      // We can't nicely format the span attributes to contain the appropriate information here,
+      // so include the whole thing as part of the server address.
+      self.attributes.rpc.serverAddress = address
+    }
+
+    switch PeerAddress(context.remotePeer) {
+    case .ipv4(let address, let port):
+      self.attributes.rpc.clientAddress = address
+      self.attributes.rpc.clientPort = port
+
+    case .ipv6(let address, let port):
+      self.attributes.rpc.clientAddress = address
+      self.attributes.rpc.clientPort = port
+
+    case .unixDomainSocket(let path):
+      self.attributes.rpc.clientAddress = path
+
+    case .other(let address):
+      self.attributes.rpc.clientAddress = address
+    }
+  }
+}
+
+private enum PeerAddress {
+  case ipv4(address: String, port: Int?)
+  case ipv6(address: String, port: Int?)
+  case unixDomainSocket(path: String)
+  case other(String)
+
+  init(_ address: String) {
+    // We expect this address to be of one of these formats:
+    // - ipv4:<host>:<port> for ipv4 addresses
+    // - ipv6:[<host>]:<port> for ipv6 addresses
+    // - unix:<uds-pathname> for UNIX domain sockets
+    let addressComponents = address.split(separator: ":", omittingEmptySubsequences: false)
+
+    guard addressComponents.count > 1 else {
+      // This is some unexpected/unknown format, so we have no way of splitting it up nicely.
+      self = .other(address)
+      return
+    }
+
+    // Check what type the transport is...
+    switch addressComponents[0] {
+    case "ipv4":
+      guard addressComponents.count == 3, let port = Int(addressComponents[2]) else {
+        // This is some unexpected/unknown format, so we have no way of splitting it up nicely.
+        self = .other(address)
+        return
+      }
+      self = .ipv4(address: String(addressComponents[1]), port: port)
```

Comment:
nit: guards are great for _early_ returns but if the return isn't actually early then `if/else` is more readable. It reduces the number of `return`s and places the positive condition next to anything that logically follows it (rather than having the negative branch in-between):

```suggestion
      if addressComponents.count == 3, let port = Int(addressComponents[2]) {
        self = .ipv4(address: String(addressComponents[1]), port: port)
      } else {
        // This is some unexpected/unknown format, so we have no way of splitting it up nicely.
        self = .other(address)
      }
      
```